specs/interop/storage-upgrade.md

```diff
@@ -0,0 +1,79 @@
+# Migrated Liquidity storage update
+
+## Overview
+
+To fully migrate liquidity from `OptimismMintableERC20` tokens to `OptimismSuperchainERC20`, the `OptimismMintableERC20Factory` needs to track its deployment history.
+
+To achieve this, a [Hash Onion](https://github.com/ethereum-optimism/design-docs/blob/main/protocol/superchain-erc20/storage-upgrade.md#2-hash-onion) solution will be used:
+
+- The `OptimismMintableERC20Factory` predeploy will include functionality to set and store the Hash Onion, as well as verify and record deployment data.
+- A Foundry script will be added to generate the Hash Onion from a list of local and remote tokens.
+- A `superchain-ops` task will be added to set the Hash Onion from the L2 `ProxyAdmin` owner.
+
+## `OptimismMintableERC20Factory` predeploy
+
+Create a new `OptimismMintableERC20FactoryInterop` contract that inherits from `OptimismMintableERC20Factory` and add the following functionality:
+
+```solidity
```

Comment:
change implementation for the one we did with Disco, that allows for array inputs